$NetBSD: patch-configure,v 1.1 2021/08/24 15:07:28 manu Exp $

Since libp11-kit installs pthread_atfork() callback, make sure
it cannot be unloaded.

Submitted upstream https://github.com/p11-glue/p11-kit/pull/383

--- configure.orig	2021-08-24 16:33:08.796020438 +0200
+++ configure	2021-08-24 16:39:51.633852995 +0200
@@ -711,8 +711,9 @@
 USE_NLS
 PKG_CONFIG_LIBDIR
 PKG_CONFIG_PATH
 PKG_CONFIG
+LDFLAGS_Z_NODELETE
 HAVE_LD_VERSION_SCRIPT_FALSE
 HAVE_LD_VERSION_SCRIPT_TRUE
 LT_SYS_LIBRARY_PATH
 OTOOL64
@@ -12747,8 +12748,38 @@
   HAVE_LD_VERSION_SCRIPT_TRUE='#'
   HAVE_LD_VERSION_SCRIPT_FALSE=
 fi
 
+# --------------------------------------------------------------------
+# Check for -z nodelete link flag
+
+SAVE_LDFLAGS="$LDFLAGS"
+LDFLAGS="$LDFLAGS -Wl,-z -Wl,nodelete"
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether linker understands -z nodelete" >&5
+printf %s "checking whether linker understands -z nodelete... " >&6; }
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+int
+main (void)
+{
+
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"
+then :
+  LDFLAGS_Z_NODELETE="-Wl,-z -Wl,nodelete"; has_option=yes
+else $as_nop
+  LDFLAGS_Z_NODELETE=""; has_option=no
+fi
+rm -f core conftest.err conftest.$ac_objext conftest.beam \
+    conftest$ac_exeext conftest.$ac_ext
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $has_option" >&5
+printf "%s\n" "$has_option" >&6; }
+LDFLAGS="$SAVE_LDFLAGS"
+
 
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
